src/core/Akka.API.Tests/CoreAPISpec.ApproveRemote.approved.txt

```diff
@@ -121,6 +121,20 @@ namespace Akka.Remote
         void Remove(T resource);
         void Reset();
     }
+    [Akka.Annotations.InternalApiAttribute()]
+    public interface IRemoteActorRefProvider : Akka.Actor.IActorRefProvider
```

Comment:
Biggest change in this PR is the introduction of the  `IRemoteActorRefProvider` interface and its adoption throughout the internals of Akka.Remote. This makes it possible to extend Akka.Remote's capabilities (such as, introducing instrumentation) without having to re-implement all of the internals.

This is an `InternalApi` and may be subject to change, but most of the consumption of this interface all occurs internally within Akka.Remote, hence why some of the changes to `Endpoints.cs` et al don't show up in the API diff. This change is binary compatible with previous versions of Akka.NET.